src/boot/efi/util.h

```diff
@@ -57,6 +57,15 @@ static inline void* xmemdup(const void *p, size_t l) {
         return memcpy(xmalloc(l), p, l);
 }
 
+static inline void **xrealloc_extra_k_items(void *p, size_t item_size, size_t current_size, size_t extra_elements) {
+        size_t new_size = (current_size + extra_elements + 1) * item_size;
+        if (current_size == 0)
+                return (void**)xrealloc(p, 0, new_size);
+        else
+                return (void**)xrealloc(p, (current_size + 1) * item_size, new_size);
+}
+
+#define xrealloc_extra_item(type, p, current_size) ((type **) xrealloc_extra_k_items(p, current_size, sizeof(type), 1))
```

Comment:
Wrong definition argument
```suggestion
#define xrealloc_extra_item(type, p, current_size) ((type **) xrealloc_extra_k_items(p, sizeof(type *), current_size, 1))
```